tests/acceptance/bootstrap/SpacesContext.php

```diff
@@ -4761,4 +4761,63 @@ public function userGetsTheDriveItemInSpace(string $user, string $file, string $
 			)
 		);
 	}
+
+	/**
+	 * @Given user :user has expired the last share of resource :resource inside of the space :spaceName
+	 *
+	 * @param string $user
+	 * @param string $resource
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 * @throws GuzzleException|JsonException
+	 */
+	public function userHasExpiredTheLastShareOfResourceInsideOfTheSpace(
+		string $user,
+		string $resource,
+		string $spaceName
+	): void {
+		$dateTime = new DateTime('yesterday');
+		$rows['expireDate'] = $dateTime->format('Y-m-d\\TH:i:sP');
+		if ($this->featureContext->isUsingSharingNG()) {
+			$space = $this->getSpaceByName($user, $spaceName);
```

Comment:
I see code duplication here, please move it to separate method. (e.g: `expireResourceShare`)